src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -743,13 +737,14 @@ else if (compare > 0)
                 if (size == 0)
                     return Optional.empty();
 
-                // TODO: Find a faster implementation using binarySearch
                 l.sort(comparator);
-                for (int i = 0; i < size; i++)
-                    if (comparator.compare(value, l.get(i)) <= 0)
-                        return Optional.of((double) i / (double) size);
-
-                return Optional.of(1.0);
+                int index = Collections.binarySearch(l, value, comparator);
+                if (index < 0)
+                    index = -index - 1;
+                else
+                    while (index > 0 && l.get(index) == l.get(index - 1))
+                        --index;

```

Comment:
@lukaseder What's your opinion on copying 6 lines above in `rankBy` and `denseRankBy`? Should I place `firstIndexOf` somewhere? I don't see any appropriate util class for that, so please advise :)